src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php

```diff
@@ -226,16 +225,23 @@ private function controlPagedResult($con, int $pageSize, string $cookie): bool
     /**
      * Retrieve LDAP pagination cookie.
      *
-     * @param resource $con
-     * @param resource $result
+     * @param resource      $con
+     * @param resource|null $result
      */
     private function controlPagedResultResponse($con, $result, string $cookie = ''): string
     {
+        $this->serverctrls = [];
+
         if (\PHP_VERSION_ID < 70300) {
             ldap_control_paged_result_response($con, $result, $cookie);
```

Comment:
What's the point of calling this function when $result === null? Is that legal for its API?

Reply:
Hum. Not sure. It will probably return an empty string. I have to admit that I never paginated LDAP result so IDK. At least I should send 0 to keep compatibility.